/ssc:
  ssc_joystick:
    ros__parameters:
      # The publish interval of the steering and speed commands (s)
      publish_interval: 0.05

      # The timeout before a lack of joystick messages will trigger a fault (s)
      joystick_fault_timeout: 3.0

      # The namespace and name of the vehicle controller ROS node (from the launch file) used to detect overrides and faults in the dbw system
      vel_controller_name: /ssc/veh_controller

      # Engage speed control module, steering will be manual unless enabled
      engage_speed_module: true

      # Engage steering control module, speed will be manual unless enabled
      engage_steering_module: true

      # The maximum speed that can be commanded (mph)
      max_speed: 15.0

      # The acceleration limit passed to the speed module (m/s^2)
      acceleration_limit: 2.0

      # The deceleration limit passed to the speed module (m/s^2)
      deceleration_limit: 2.5

      # The maximum deceleration limit inÂ (m/s^2)
      max_deceleration_limit: 4.0

      # The exponent to control the shape/modulation of the steering command, needs to be >= 1
      steering_exponent: 2.5

      # The maximum amount of curvature able to be set (1/m)
      max_curvature: 0.12

      # The maximum curvature rate passed to the steering module (1/m/ms)
      max_curvature_rate: 0.1

      # Added for testing quick brake condition, quick_brake_speed needs to be > 0 and related mode parameters in SSC's configuration need to be non-zero
      test_quick_brake: false

      # The speed limit beyond which zero speed is requested with a zero decel
      quick_brake_speed: 0.0


      ### Controller Parameters ###

      ### Buttons
      # The joystick button used to engage the joystick controller
      engage1_button: 6

      # The joystick button used to disengage the joystick controller
      engage2_button: 7

      # The joystick button used to change the gear to park
      park_button: 3

      # The joystick button used to change the gear to neutral
      neutral_button: 2

      # The joystick button used to change the gear to drive
      drive_button: 0

      # The joystick button used to change the gear to reverse
      reverse_button: 1

      # The joystick button used to command the right turn signal on
      right_turn_button: 5

      # The joystick button used to command the left turn signal on
      left_turn_button: 4

      # The joystick axes use to provide speed up and slow down commands
      speed_axes: 7

      # The sign of the speed axes to control which is speed up and which is slow down
      speed_up_sign: 1.0

      # How much the speed should increase or decrease with each button press (mph)
      speed_step: 1.0

      ### Axes
      # The joystick axis use to control the desired curvature
      steering_axes: 3

      # The sign of the joystick to control left and right
      steering_sign: 1.0

      # The joystick axis used to step the curvature to the left or right
      steer_btn_axes: 6

      # Determines which button steps positive (to the left) and negative (to the right)
      steer_btn_sign: 1

      # How much to increment or decrement the curvature with each button press
      steer_btn_step: 0.01

      # The joystick axis used for override braking
      brake_axes: 2

      # The sign of the axis used for override braking
      brake_sign: 1.0
